src/eko/runner/legacy.py

```diff
@@ -0,0 +1,124 @@
+"""Main application class of eko."""
+import copy
+import logging
+import os
+from typing import Union
+
+import numpy as np
+
+from .. import interpolation, msbar_masses
+from ..couplings import Couplings, couplings_mod_ev
+from ..evolution_operator.grid import OperatorGrid
+from ..io import EKO, Operator, runcards
+from ..io.types import QuarkMassSchemes, RawCard
+from ..thresholds import ThresholdsAtlas
+from . import commons
+
+logger = logging.getLogger(__name__)
+
+
+class Runner:
+    """Represents a single input configuration.
+
+    For details about the configuration, see :doc:`here </code/IO>`
+
+    Attributes
+    ----------
+    setup : dict
+        input configurations
+
+    """
+
+    banner = commons.BANNER
+
+    def __init__(
+        self,
+        theory_card: Union[RawCard, runcards.TheoryCard],
+        operators_card: Union[RawCard, runcards.OperatorCard],
+        path: os.PathLike,
+    ):
+        """Initialize runner.
+
+        Parameters
+        ----------
+        theory_card: dict
+            theory parameters and options
+        operators_card: dict
+            operator specific options
+
+        """
+        new_theory, new_operator = runcards.update(theory_card, operators_card)
+
+        # Store inputs
+        self.path = path
+        self._theory = new_theory
+
+        # setup basis grid
+        bfd = interpolation.InterpolatorDispatcher(
+            xgrid=new_operator.rotations.xgrid,
+            polynomial_degree=new_operator.configs.interpolation_polynomial_degree,
+        )
+
+        # setup the Threshold path, compute masses if necessary
+        masses = None
+        if new_theory.quark_masses_scheme is QuarkMassSchemes.MSBAR:
+            masses = msbar_masses.compute(new_theory)
+        elif new_theory.quark_masses_scheme is QuarkMassSchemes.POLE:
+            masses = tuple(mq.value**2 for mq in new_theory.quark_masses)
+        else:
+            raise ValueError(f"Unknown mass scheme '{new_theory.quark_masses_scheme}'")
+
+        # call explicitly iter to explain the static analyzer that is an
+        # iterable
+        thresholds_ratios = list(iter(new_theory.matching))
+        tc = ThresholdsAtlas(
+            masses=masses,
+            q2_ref=new_operator.mu20,
+            nf_ref=new_theory.num_flavs_init,
+            thresholds_ratios=thresholds_ratios,
+            max_nf=new_theory.num_flavs_max_pdf,
+        )
+
+        # strong coupling
+        sc = Couplings(
+            couplings_ref=np.array(new_theory.couplings.values),
+            scale_ref=new_theory.couplings.alphas.scale**2,
+            thresholds_ratios=thresholds_ratios,
+            masses=tuple(m2 / new_theory.fact_to_ren**2 for m2 in masses),
+            order=new_theory.order,
+            method=couplings_mod_ev(new_operator.configs.evolution_method.value),
+            nf_ref=new_theory.num_flavs_ref,
+            max_nf=new_theory.num_flavs_max_as,
```

Comment:
why are `num_falvs_ref` and `num_flavs_max_as` not inside `new_theory.couplings`?

Reply:
Just because I initially started couplings with only the reference values of `alphas` and `alphaem`, later on I added reference scales (since it made sense to keep them together).
We can also move them there, since they are only used for couplings, and we can pass the full data struct to the `Couplings` object :)

Reply:
Now an item on the original comment https://github.com/NNPDF/eko/pull/172#issue-1477202955, I will do.
I close here, just because the information is available in a more visible place.